src/PSR7/Validators/Headers.php

```diff
@@ -20,22 +19,14 @@ class Headers
      */
     public function validate(MessageInterface $message, array $headerSpecs) : void
     {
-        $messageHeaders = $message->getHeaders();
+        // By default this will not report unexpected headers (soft validation)
```

Comment:
Leaving this commit with removed header iteration does no sence, since implementing todo will mean revert this change back